mlkem/polyvec.c

```diff
@@ -27,8 +27,7 @@ void polyvec_compress(uint8_t r[KYBER_POLYVECCOMPRESSEDBYTES],
   for (i = 0; i < KYBER_K; i++) {
     for (j = 0; j < KYBER_N / 8; j++) {
       for (k = 0; k < 8; k++) {
-        t[k] = a->vec[i].coeffs[8 * j + k];
-        t[k] += ((int16_t)t[k] >> 15) & KYBER_Q;
+        t[k] = scalar_signed_to_unsigned_q_16(a->vec[i].coeffs[8 * j + k]);
```

Comment:
Let's not mix functional changes with the changes needed to address the newly added warnings.

Reply:
The original code convert signed to unsigned, I found the `scalar_signed_to_unsigned_q_16` is the perfect function to do so. 
Strange, in poly.c we have bound check and proof for `poly_compress`: 

```c
void poly_compress(uint8_t r[KYBER_POLYCOMPRESSEDBYTES], const poly *a) {
  POLY_UBOUND(a, KYBER_Q);
  uint8_t t[8] = {0};

  const int num_blocks = KYBER_N / 8;

#if (KYBER_POLYCOMPRESSEDBYTES == 128)
  for (int i = 0; i < num_blocks; i++)
      // clang-format off
        ASSIGNS(i, OBJECT_WHOLE(t), OBJECT_WHOLE(r))
        INVARIANT(i >= 0 && i <= num_blocks)
        DECREASES(num_blocks - i)
    // clang-format on
    {
      for (int j = 0; j < 8; j++).
...
```
However, the `polyvec_compress` isn't protected the same way as `poly_compress`, although it has division resistant code.

Reply:
revert in https://github.com/pq-code-package/mlkem-c-aarch64/pull/248/commits/89089b6c7e90725904417001c5b17f2352ed6e1d